src/main/typescript/data/item/weapon/attack/properties.ts

```diff
@@ -1,60 +1,81 @@
-      this.attacks[name] = new AttackProperties(name, source, owningWeapon);
-    });
-  }
-
+export type AttacksProperties = AttacksSource & {
   /** The Weapon Attacks, created from the sources */
-  attacks: Record<string, AttackProperties> = {};
+  attacks: Record<string, AttackProperties>;
 
   /** Get all attacks matching the given tags. */
-  getMatching(tags: string[] | undefined): AttackProperties[] {
-    return Object.values(this.attacks).filter((attack) => attack.matches(tags));
-  }
+  getMatching(tags: string[] | undefined): AttackProperties[];
 
   /**
    * Apply a skill damage dice modifier to all attacks, if the attack qualifies
    * for it. This is based on the skill of the weapon and the skill value of the
    * given Actor.
    */
-  applySkillDamageDiceMod(actor: WvActor | null, weapon: Weapon): void {
-    if (!actor) return;
-
-    Object.values(this.attacks).forEach((attack) =>
-      attack.damage.applySkillDamageDiceMod(actor, weapon)
-    );
-  }
+  applySkillDamageDiceMod(actor: WvActor | null, weapon: Weapon): void;
 
   /**
    * Apply a Strength damage dice modifier to all attacks, based on the Strength
    * of the given Actor.
    */
-  applyStrengthDamageDiceMod(actor: WvActor | null): void {
-    if (!actor) return;
+  applyStrengthDamageDiceMod(actor: WvActor | null): void;
+};
 
-    Object.values(this.attacks).forEach((attack) =>
-      attack.damage.applyStrengthDamageDiceMod(actor)
-    );
+export const AttacksProperties = {
+  from(source: AttacksSource, owningWeapon: Weapon): AttacksProperties {
+    const attacks: Record<string, AttackProperties> = {};
+    Object.entries(source.sources).forEach(([name, source]) => {
+      attacks[name] = new AttackProperties(name, source, owningWeapon);
+    });
+    return {
+      ...source,
+      attacks,
+      getMatching(tags: string[] | undefined): AttackProperties[] {
+        return Object.values(this.attacks).filter((attack) =>
+          attack.matches(tags)
+        );
+      },
+
+      /**
+       * Apply a skill damage dice modifier to all attacks, if the attack qualifies
+       * for it. This is based on the skill of the weapon and the skill value of the
+       * given Actor.
+       */
+      applySkillDamageDiceMod(actor: WvActor | null, weapon: Weapon): void {
+        if (!actor) return;
+
+        Object.values(this.attacks).forEach((attack) =>
+          attack.damage.applySkillDamageDiceMod(actor, weapon)
+        );
+      },
+
+      /**
+       * Apply a Strength damage dice modifier to all attacks, based on the Strength
+       * of the given Actor.
+       */
+      applyStrengthDamageDiceMod(actor: WvActor | null): void {
+        if (!actor) return;
+
+        Object.values(this.attacks).forEach((attack) =>
+          attack.damage.applyStrengthDamageDiceMod(actor)
+        );
+      }
+    };
   }
-}
+};
 
-export class AttackProperties extends AttackSource {
+export class AttackProperties implements AttackSource {
```

Comment:
Here it works because
- It's a one-off thing and I already implemented it like that
- There aren't many attributes to implement
In bigger types, there's a lot fields to re-declare, and if any of them are optional the Typechecking errors get really cryptic really fast.